### ISSUE TITLE
mkosi fails while spawning systemd-nspawn during bios-enabled build

### ISSUE THREAD

**Issue by mactijn** [systemd/mkosi#403]
While attempting to build a Debian/stable image on a fully up-to-date Ubuntu 18.04 host, and using the `master` branch (3a3a21e) of mkosi, a fatal error occurs (see below). The cause seems to be a lack of the `--console=` option for systemd-nspawn (part of Ubuntu's `systemd-container` package).

An 18.04 install comes with (a somewhat modified) systemd v237, with `237-3ubuntu10.39` as the version string for the latest available systemd packages. According to the man page, this should be fine:
> The current version requires systemd 233 (or actually, systemd-nspawn of it).

However, after some digging I found that the `--console=` option was only added in systemd v241-rc1.

I was contemplating just submitting a PR to update the minimal systemd version requirement in the man page, however as 18.04 is (at least from my perspective) still quite a popular choice due to its LTS status, it occurred to me that this might perhaps not be so well-received by everyone (though it would fit with the "legacy-free" ideology).

The error:
```
[...]
‣ Installing Debian complete.
‣ Assigning hostname...
‣ Assigning hostname complete.
‣ Installing boot loader...
systemd-nspawn: unrecognized option '--console=pipe'
‣ Unmounting Package Cache...
‣ Unmounting Package Cache complete.
‣ Unmounting image...
‣ Unmounting image complete.
‣ Detaching image file...
‣ Detaching image file complete.
Traceback (most recent call last):
  File "../../Projects/mkosi/mkosi", line 4872, in <module>
    main()
  File "../../Projects/mkosi/mkosi", line 4868, in main
    run_verb(a)
  File "../../Projects/mkosi/mkosi", line 4843, in run_verb
    build_stuff(args)
  File "../../Projects/mkosi/mkosi", line 4667, in build_stuff
    raw, tar, root_hash = build_image(args, workspace, do_run_build_script=False, cleanup=True)
  File "../../Projects/mkosi/mkosi", line 4479, in build_image
    install_boot_loader(args, workspace.name, loopdev, cached)
  File "../../Projects/mkosi/mkosi", line 2393, in install_boot_loader
    install_boot_loader_debian(args, workspace, loopdev)
  File "../../Projects/mkosi/mkosi", line 2335, in install_boot_loader_debian
    install_grub(args, workspace, loopdev, "grub")
  File "../../Projects/mkosi/mkosi", line 2301, in install_grub
    loopdev, nspawn_params=nspawn_params)
  File "../../Projects/mkosi/mkosi", line 1215, in run_workspace_command
    run(cmdline, check=True)
  File "../../Projects/mkosi/mkosi", line 76, in run
    return subprocess.run(cmdline, **kwargs)
  File "/usr/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['systemd-nspawn', '--quiet', '--directory=/var/tmp/mkosi-z05sa23k/root', '--uuid=a7fc552fd1dd48b9bbb0b425f716fe89', '--machine=mkosi-ec94bf0a0ed14195af3732bc27264ac8', '--as-pid2', '--register=no', '--bind=/var/tmp/mkosi-z05sa23k/var-tmp:/var/tmp', '--setenv=SYSTEMD_OFFLINE=1', '--private-network', '--bind-ro=/dev', '--property=DeviceAllow=/dev/loop2', '--console=pipe', '--property=DeviceAllow=/dev/loop2p3', '--', 'grub-install', '--modules=ext2 part_gpt', '--target=i386-pc', '/dev/loop2']' returned non-zero exit status 1.
```
Command used:
```
$ sudo ../../Projects/mkosi/mkosi --distribution debian --release stable \
  --bootable --boot-protocol bios --format gpt_ext4 --hostname foobarbaz \
  --cache=./cache/pkg --with-network --swap-size 1G --password foobar \
  --kernel-command-line 'console=ttyS0'
```

I haven't yet checked into the workings of the `--console=` option of `systemd-nspawn`,  but given the terminology involved I was wondering if this is perhaps solvable with an external helper?

**Comment by behrmann** [comment]
Quickly looking at the blame output of `mkosi` I find that there is only one line with `--console=` and that was introduced by b0174ea179ad8622dc404df20bfc139973bb249e to fix #380. Without having dug into this any further it might be, that it is a slight impasse here, that could be solved by changing the commandline depending on the target distro.

I do agree that it is sensible to support at least the current LTS of Ubuntu, which 18.04 still is.

**Comment by DaanDeMeyer** [comment]
Fixed in master
